vault/diagnose/storage_checks.go

```diff
@@ -14,66 +14,55 @@ const (
 	secretKey string = "diagnose"
 	secretVal string = "diagnoseSecret"
 
-	timeOutErr        string = "storage call timed out after 20 seconds: "
-	DirAccessErr      string = "consul storage does not connect to local agent, but directly to server"
-	AddrDNExistErr    string = "config address does not exist: 127.0.0.1:8500 will be used"
-	wrongRWValsPrefix string = "Storage get and put gave wrong values: "
+	LatencyWarning    string        = "latency above 100 ms for storage call: "
+	DirAccessErr      string        = "consul storage does not connect to local agent, but directly to server"
+	AddrDNExistErr    string        = "config address does not exist: 127.0.0.1:8500 will be used"
+	wrongRWValsPrefix string        = "Storage get and put gave wrong values: "
+	latencyThreshold  time.Duration = time.Millisecond * 100
 )
 
-// StorageEndToEndLatencyCheck calls Write, Read, and Delete on a secret in the root
-// directory of the backend.
-// Note: Just checking read, write, and delete for root. It's a very basic check,
-// but I don't think we can necessarily do any more than that. We could check list,
-// but I don't think List is ever going to break in isolation.
-func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error {
-
-	c2 := make(chan error)
-	go func() {
-		err := b.Put(context.Background(), &physical.Entry{Key: secretKey, Value: []byte(secretVal)})
-		c2 <- err
-	}()
-	select {
-	case errOut := <-c2:
-		if errOut != nil {
-			return errOut
-		}
-	case <-time.After(20 * time.Second):
-		return fmt.Errorf(timeOutErr + "operation: Put")
+func EndToEndLatencyCheckWrite(ctx context.Context, uuid string, b physical.Backend) error {
+	start := time.Now()
+	err := b.Put(context.Background(), &physical.Entry{Key: secretKey, Value: []byte(secretVal)})
+	duration := time.Since(start)
+	if err != nil {
+		return err
 	}
+	if duration > latencyThreshold {
+		return fmt.Errorf(LatencyWarning + "operation: put")
```

Comment:
Per comment above, it'd be nice to have some structure and actually include the latency.  It'll be helpful to folks to report if the latency is 101ms or 3000ms.

Reply:
Will do!